packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -1891,6 +1892,12 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin {
     if (removeBottomInset)
       data = data.removeViewInsets(removeBottom: true);
 
+    if(maintainBottomViewPadding && data.viewInsets != EdgeInsets.zero){
```

Comment:
should this only check that viewInsets.bottom is not zero? (In other words: why should a top viewInsets change the bottom padding?